lnwallet/wallet.go

```diff
@@ -720,100 +800,82 @@ func (l *LightningWallet) handleContributionMsg(req *addContributionMsg) {
-	// transaction which will allow opening a lightning channel.
-	pendingReservation.fundingTx = wire.NewMsgTx(1)
-	fundingTx := pendingReservation.fundingTx
-
 	// Some temporary variables to cut down on the resolution verbosity.
 	pendingReservation.theirContribution = req.contribution
 	theirContribution := req.contribution
 	ourContribution := pendingReservation.ourContribution
 
-	// Add all multi-party inputs and outputs to the transaction.
-	for _, ourInput := range ourContribution.Inputs {
-		fundingTx.AddTxIn(ourInput)
-	}
-	for _, theirInput := range theirContribution.Inputs {
-		fundingTx.AddTxIn(theirInput)
-	}
-	for _, ourChangeOutput := range ourContribution.ChangeOutputs {
-		fundingTx.AddTxOut(ourChangeOutput)
-	}
-	for _, theirChangeOutput := range theirContribution.ChangeOutputs {
-		fundingTx.AddTxOut(theirChangeOutput)
-	}
-
-	ourKey := pendingReservation.ourContribution.MultiSigKey
-	theirKey := theirContribution.MultiSigKey
-
-	// Finally, add the 2-of-2 multi-sig output which will set up the lightning
-	// channel.
-	channelCapacity := int64(pendingReservation.partialState.Capacity)
-	witnessScript, multiSigOut, err := input.GenFundingPkScript(
-		ourKey.PubKey.SerializeCompressed(),
-		theirKey.PubKey.SerializeCompressed(), channelCapacity,
+	var (
+		chanPoint *wire.OutPoint
+		err       error
 	)
-	if err != nil {
-		req.err <- err
-		return
-	}
-
-	// Sort the transaction. Since both side agree to a canonical ordering,
-	// by sorting we no longer need to send the entire transaction. Only
-	// signatures will be exchanged.
-	fundingTx.AddTxOut(multiSigOut)
-	txsort.InPlaceSort(pendingReservation.fundingTx)
 
-	// Next, sign all inputs that are ours, collecting the signatures in
-	// order of the inputs.
-	pendingReservation.ourFundingInputScripts = make([]*input.Script, 0,
-		len(ourContribution.Inputs))
-	signDesc := input.SignDescriptor{
-		HashType:  txscript.SigHashAll,
-		SigHashes: txscript.NewTxSigHashes(fundingTx),
-	}
-	for i, txIn := range fundingTx.TxIn {
-		info, err := l.FetchInputInfo(&txIn.PreviousOutPoint)
-		if err == ErrNotMine {
-			continue
-		} else if err != nil {
-			req.err <- err
+	// At this point, we can now construct our channel point. Depenindg on
+	// which type of intent we obtained from our chanfunding.Assembler,
+	// we'll carry out a distinct set of steps.
+	switch fundingIntent := pendingReservation.fundingIntent.(type) {
+	case *chanfunding.ShimIntent:
+		chanPoint, err = fundingIntent.ChanPoint()
+		if err != nil {
+			req.err <- fmt.Errorf("unable to obtain chan point: %v", err)
 			return
 		}
 
-		signDesc.Output = &wire.TxOut{
-			PkScript: info.PkScript,
-			Value:    int64(info.Value),
-		}
-		signDesc.InputIndex = i
+		pendingReservation.partialState.FundingOutpoint = *chanPoint
+
+	case *chanfunding.FullIntent:
+		// Now that we know their public key, we can bind theirs as
+		// well as ours to the funding intent.
+		fundingIntent.BindKeys(
```

Comment:
a safer way of doing this could be to provide the keys as parameters to `CompileFundingTx()`

Reply:
I considered that, but then it wouldn't be possible to obtain the funding output in isolation without also constructing+signing the funding transaction.

Reply:
The method as is signs the funding transaction. For the upcoming PSBT integration, we'll need just the output itself to insert into the partial transaction for signing. As far as the current state, the docs are clear that one it to interact with the object in a builder-like format. A similar format will be used for the PSBT integration (get unsigned, sign our portion, combine with remote party's sig, etc).